### PR TITLE
Default `ENABLE_LAPACK` as `OFF`

### PR DESCRIPTION
**Pull request by multiphaseCFD**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

[SC-70045]

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,9 @@
 
 ### Improvements
 
+* Default `ENABLE_LAPACK` as `OFF` for `lightning` backends build.
```
Comment (vincentmr) [inline .github/CHANGELOG.md:18]:
```suggestion
* `ENABLE_LAPACK` is off by default for all Lightning backends.
```

File: .github/workflows/tests_lgpumpi_cpp.yml [hunk 2]
```diff
@@ -44,6 +44,7 @@ jobs:
       max-parallel: 1
       matrix:
         mpilib: ["mpich", "openmpi"]
+        enable_lapack: ["OFF", "ON"]
```
Comment (vincentmr) [inline .github/workflows/tests_lgpumpi_cpp.yml:47]: This workflow is already quite demanding, do we really need to test both with and without LAPACK given that the non-MPI code already tests that.

**Review by vincentmr — COMMENTED** [review]
Not much to complain about, thanks @multiphaseCFD .

File: .github/workflows/tests_lgpu_cpp.yml [hunk 3]
```diff
@@ -61,6 +61,7 @@ jobs:
     strategy:
       matrix:
         pl_backend: ["lightning_gpu"]
+        enable_lapack: ["OFF", "ON"]
         cuda_version: ["12"]
 
     name: C++ Tests (${{ matrix.pl_backend }}, cuda-${{ matrix.cuda_version }})
```
Comment (maliasadi) [inline .github/workflows/tests_lgpu_cpp.yml:67]:
```suggestion
        enable_lapack: ["OFF", "ON"]
        cuda_version: ["12"]

    name: C++ Tests (${{ matrix.pl_backend }}, cuda-${{ matrix.cuda_version }}, enable_lapack=${{ matrix.enable_lapack }})
```

File: .github/workflows/tests_linux_cpp.yml [hunk 4]
```diff
@@ -48,6 +48,7 @@ jobs:
         pl_backend: ["lightning_qubit"]
         enable_kernel_omp: ["OFF", "ON"]
         enable_kernel_avx_streaming: ["OFF", "ON"]
+        enable_lapack: ["OFF", "ON"]
```
Comment (maliasadi) [inline .github/workflows/tests_linux_cpp.yml:51]: Could you update Line 56?

File: .github/workflows/tests_lkcuda_cpp.yml [hunk 5]
```diff
@@ -45,6 +45,7 @@ jobs:
         os: [ubuntu-22.04]
         exec_model: ["CUDA"]
         kokkos_version: ["4.3.01"]
+        enable_lapack: ["OFF", "ON"]
```
Comment (maliasadi) [inline .github/workflows/tests_lkcuda_cpp.yml:48]: And line 116?

**Review by maliasadi — COMMENTED** [review]
Thanks @multiphaseCFD!

File: .github/workflows/tests_linux_cpp.yml [hunk 6]
```diff
@@ -48,11 +48,12 @@ jobs:
...
         exclude:
         - enable_kernel_omp: OFF
           enable_kernel_avx_streaming: ON
     timeout-minutes: 60
-    name: C++ Tests (${{ matrix.pl_backend }}, ENABLE_KERNEL_OMP=${{ matrix.enable_kernel_omp }}, ENABLE_KERNEL_AVX_STREAMING=${{ matrix.enable_kernel_avx_streaming }})
+    name: C++ Tests (${{ matrix.pl_backend }}, ENABLE_KERNEL_OMP=${{ matrix.enable_kernel_omp }}, ENABLE_KERNEL_AVX_STREAMING=${{ matrix.enable_kernel_avx_streaming }}), ENABLE_LAPACK=${{ matrix.enable_lapack }})
```
Comment (maliasadi) [inline .github/workflows/tests_linux_cpp.yml:56]: Not related to your PR, but just noticed that we can make this consistent with other names too,
```suggestion
    name: C++ Tests (${{ matrix.pl_backend }}, enable_kernel_omp=${{ matrix.enable_kernel_omp }}, enable_kernel_avx_streaming=${{ matrix.enable_kernel_avx_streaming }}), enable_lapack=${{ matrix.enable_lapack }})
```

**Review by maliasadi — APPROVED** [review]

**Review by vincentmr — APPROVED** [review]
LGTM, thanks @multiphaseCFD !

**Pull request merged**